File: Data Structures/q.py




# class Node:
# 	def __init__(self, value):
# 		self.value = value
# 		self.next = None

# class Queue:
# 	def __init__(self):
# 		self.first = None
# 		self.last = None
# 		self.length = 0

# 	def peek(self):
# 		return self.first

# 	def isEmpty(self):
# 		return bool(self.length is 0)

# 	def nq(self,value):
# 		newNode = Node(value)
# 		if (self.length == 0):
# 			self.first = newNode
# 			self.last = newNode
# 		else:
# 			self.last.next = newNode
# 			self.last = newNode
# 		self.length += 1
# 		return self

# 	def dq(self):
# 		if (self.length <= 0):
# 			return None
# 		elif (self.length == 1):
# 			dq = self.first
# 			self.first = None
# 			self.last = None
# 		else:
# 			dq = self.first
# 			self.first = self.first.next

# 		self.length -= 1
# 		return dq



# myQueue = Queue()

File: Data Structures/q.js
class Node {
	constructor(value) {
		this.value = value
		this.next = null
	}
}

class Queue{
	constructor() {
		this.first = null
		this.last = null
		this.length = 0
	}
	peek() {
		return this.first
	}
	nq(value) {
		let newNode = new Node(value)
		if (this.length === 0) {
			this.first = newNode
			this.last = this.first
		} else {
			this.last.next = newNode
			this.last = newNode
		}
		this.length++
		return this
	}
	dq() {
		if (this.length === 0) {
			return "Nothing to dq"
		}

		let popped
		if (this.length === 1) {
			popped = this.first
			this.first = null
			this.last = null
		} else {
			popped = this.first
			this.first = this.first.next
		}
		this.length--
		return popped
	}
	isEmpty() {
		return Boolean(!this.length)
	}
}

let myQueue = new Queue()